moto/iot/models.py

```diff
@@ -857,8 +857,30 @@ def delete_thing_group(self, thing_group_name, expected_version):
         del self.thing_groups[thing_group.arn]
 
     def list_thing_groups(self, parent_group, name_prefix_filter, recursive):
-        thing_groups = self.thing_groups.values()
-        return thing_groups
+        if recursive is None:
+            recursive = True
+        if name_prefix_filter is None:
+            name_prefix_filter = ""
+        if parent_group and parent_group not in [
+            _.thing_group_name for _ in self.thing_groups.values()
+        ]:
+            raise ResourceNotFoundException()
+        thing_groups = [
+            _ for _ in self.thing_groups.values() if _.parent_group_name == parent_group
```

Comment:
The `parent_group` argument is not mandatory - what happens if it's None? This if-statement reads like `thing_groups` would become an empty list - when it should return everything if parent_group is not supplied.

Reply:
Its actually allowed for it to be None. When `parent_group` is None `thing_groups` would be the list of groups which have no parents. That is, the full list of groups on the first level of the tree. Now, as `recursive` is true by default, each of those groups return the data for the childs recursively, returning all the groups in the end. 
But don't take my word for it, its actually on a test case 😃

Reply:
Ah, gotcha. I saw the test case, but couldn't quite figure out how it worked :)